docs/docusaurus/docs/core/connect_to_data/sql_data/_create_a_data_source/_create_a_data_source.md

```diff
@@ -54,8 +54,35 @@ import DatasourceMethodReferenceTable from './_datasource_method_reference_table
 
    ```python title="Python" name="docs/docusaurus/docs/core/connect_to_data/sql_data/_create_a_data_source/postgres.py create data source"
    ```
+4. Optional. If you're connecting to Snowflake and want to use key-pair authentication instead of a password, pass the private key with `kwargs`. Note that a placeholder password is still required to pass the configuration validation, but the password will not be used if a `private_key` is provided.
+
+   ```python title="Python"
+   # For details on how to access your private key, refer to "Configure credentials" above 
```

Comment:
Note for reviewer: I didn't try to put this code under test because it's for a cloud context and https://greatexpectations.atlassian.net/browse/CORE-335 is blocking cloud context docs snippet tests